src/integer_mod_q/mat_zq/arithmetic/mul.rs

```diff
@@ -51,6 +53,55 @@ impl Mul for &MatZq {
     }
 }
 
+impl Mul<&MatZ> for &MatZq {
+    type Output = MatZq;
+
+    /// Implements the [`Mul`] trait for [`MatZq`] and [`MatZ`].
+    ///
+    /// [`Mul`] is implemented for any combination of owned and borrowed [`MatZq`] and [`MatZ`].
+    ///
+    /// Parameters:
+    /// - `other`: specifies the value to multiply with `self`
+    ///
+    /// Returns the product of `self` and `other` as a [`MatZq`].
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::MatZq;
+    /// use qfall_math::integer::MatZ;
+    /// use std::str::FromStr;
+    ///
+    /// let a = MatZq::from_str("[[2,1],[1,2]] mod 3").unwrap();
+    /// let b = MatZ::from_str("[[1,0],[0,1]]").unwrap();
+    ///
+    /// let c = &a * &b;
+    /// let d = a * b;
+    /// let e = &c * d;
+    /// let f = c * &e;
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Panics if the dimensions of `self` and `other` do not match for multiplication.
+    fn mul(self, other: &MatZ) -> Self::Output {
```

Comment:
Previously we always added mul_safe as a function that does not panic. Is there any reason you did it differently?

Reply:
We would need mul_safe for both directions (or a macro for doing so). I thought that this might be unnecessary. But if you think we could need it, I can add it as well.